src/main/java/org/jenkinsci/plugins/workflow/libs/LibraryAdder.java

```diff
@@ -190,14 +190,15 @@
 
             if(versionCacheDir.exists()) {
                 listener.getLogger().println("Library " + name + "@" + version + " is cached. Copying from home.");
-                lastReadFile.touch(System.currentTimeMillis());
             } else {
                 listener.getLogger().println("Caching library " + name + "@" + version);
                 versionCacheDir.mkdirs();
                 retrieveLockFile.touch(System.currentTimeMillis());
                 retriever.retrieve(name, version, changelog, versionCacheDir, run, listener);
                 retrieveLockFile.delete();
             }
+            lastReadFile.touch(System.currentTimeMillis());
+            versionCacheDir.withSuffix("-name.txt").write(name, "UTF-8");
```

Comment:
We write this every time rather than just when the cache is first created to handle caches created after the security fix but before this PR.